packages/amplify-cli/src/init-steps/preInitSetup.ts

```diff
@@ -27,8 +27,11 @@ export const preInitSetup = async (context: $TSContext): Promise<$TSContext> =>
  *
  * @throws error if url is not a valid remote github url
  */
-const validateGithubRepo = async (repoUrl: string): Promise<void> => {
+function validateGithubRepo(repoUrl: string | boolean): asserts repoUrl is string {
```

Comment:
We made a decision in the past to convert standard functions to arrow functions. Is there a reason for converting it back?

Reply:
[Assertion ](https://github.com/aws-amplify/amplify-cli/pull/11962#discussion_r1108848638) functions in typescript cannot use arrow syntax. I converted this to an assertion function to make the code in the calling function simpler. 

Why an assertion function here? `repoUrl` is one of those odd properties that is misused in the application. There is a single line of code that sets the property to `true` instead of a string. Therefore, the value can either be `true`, or a url to a git repo. In order to avoid type casting (potentially erroneously) in the calling method, I refactored this validation method to have a typescript assertion return value. This makes it so that we don't need to use coercion in the caller to call repoUrl a string.

Reply:
https://github.com/sdstolworthy/amplify-cli/blob/4b0076b29146e57609057d5ca870ed049cbc21f6/packages/amplify-cli/src/init-steps/postInitSetup.ts#L19

Reply:
Makes perfect sense, thanks for the explanation! Also, TIL about assertion functions in TS: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions